### PR TITLE
Change operational requirements from model to enum

### LINKED ISSUE #2212: Refactor OperationalRequirements to be an Enum, instead of a database table

### PR DESCRIPTION
**Pull request by petertgiles**
This branch changes the operational requirements from a full Laravel model to a graphql enum.  It includes model updates, schema/api updates, test updates, frontend updates, and translation updates.

Suggested use cases for testing during review:

**talent search**:
- [ ] create search request
- [ ] submit search request

**admin**:

- [ ] review search request
- [ ] create pool
- [ ] update pool
- [ ] create pool candidate
- [ ] update pool candidate
- [ ] update user

Closes #2212

### CODE REVIEW

File: api/database/helpers/EnumsForFactories.php
Comment (petertgiles) [inline api/database/helpers/EnumsForFactories.php:5]: I don't love this but I like it better than pasting string literals in all the factories and seeders.  Any way to pull it from the schema?  I can create an issue to investigate.

File: api/tests/Unit/PoolCandidateTest.php
Comment (petertgiles) [inline api/tests/Unit/PoolCandidateTest.php:7]: See issue #2364 for background on a lot of changes in this file.

**Comment by vd1992** [comment]
Whew, big!
I'll definitely wait until this is merged in for pr #2369 
It calls operational requirements

**Review by yonikid15 — APPROVED** [review]
Wow you knocked this out in record time 😲 I found no issues on my end!

**Comment by petertgiles** [comment]
Created https://github.com/GCTC-NTGC/gc-digital-talent/issues/2400

**Pull request merged**
